workspace/__lib__/xod/stream/accumulate-string/patch.cpp

```diff
@@ -0,0 +1,38 @@
+
+struct State {
+    uint8_t* buff;
+    size_t cursor;
```

Comment:
You can make the node slightly more efficient if replace the meaning of `cursor` from “index” to “pointer”. That is, make `cursor` a `char*` pointing to the exact position in `buff`.